src/mosm.sol

```diff
@@ -54,9 +54,21 @@ contract Mosm is LibNote {
     // Whitelisted feed readers, set by an auth
     mapping (bytes32 => mapping (address => uint256)) public bud;
 
+    address[] public lifted;
+    function lifted_length() public view returns(uint256) { return lifted.length; }
+
+    address[] public authedMedian;
+    function authedMedian_length() public view returns(uint256) { return authedMedian.length; }
+
+    address[] public authedOsm;
+    function authedOsm_length() public view returns(uint256) { return authedOsm.length; }
+
 // contract Median {
     // --- Auth ---
-    function median_rely(address usr) external note median_auth { wards["median"][usr] = 1; }
+    function median_rely(address usr) external note median_auth {
+        wards["median"][usr] = 1;
+        authedMedian.push(usr);
```

Comment:
As there is no check whether the `usr` address is already `auth`-ed, two small things to note here:
1. The function is not idempotent anymore because an `auth`-ed address will again be added to the array. I'm not sure how much the backend depends on this property though.
2. It reduces the amount of information the addresses in the array itself carry. Being certain an address is only added to the array at the moment they received `auth` gives the auditor assurance that for every address being twice in the array, it must have lost `auth` at some point in between the two additions. With the current implementation, an address being more than once in the array could have either lost and regained `auth` **or** maybe `rely()` was just called twice with the same address.

Note that the `osm_rely` function is equally affected.

Reply:
yeag, both points are true. per 2), this was by design, since we don't particularly care about the conditions under which a `usr` has been authed. we just want to ensure that all additions are auditable and revocable. primarily we're concerned with revocation. after we deploy a median contract, Maker governance needs to revoke all of our permissions, and we currently have to supply them with the addresses, which is error-prone and could be manipulated.

we could put a check here to prevent re-auth I suppose. I see marginal benefits to both ways. @NiklasKunkel thoughts?

Reply:
The frequency of calls to `rely` and `deny` is expected to be low (single digits). I don't mind adding the check at a slight gas penalty if it makes the contract easier for an auditor to reason about.

Reply:
updated, idempotency restored...